### DISCUSSION TITLE
Bar Charts X and Y labels formatting and customization

### DISCUSSION THREAD

**Discussion by dpedwards** [Budibase/budibase#5065] [Help] [answered]
Hello is it possible to format the data for the X and Y labels somehow so that the scale on the axes fits correctly? 
_Thank you in advance for any response_ 

#1 Formatting option does not work:
![no_own_data_on_xy_axes-Budibase6](https://user-images.githubusercontent.com/2211088/159886103-e2fbfa88-5428-47e2-ac68-a6e3e6dea023.png)

#2 Axes do not scale correctly:
![no_own_data_on_xy_axes-Budibase5](https://user-images.githubusercontent.com/2211088/159885019-3cb90547-c203-47c2-a4d3-62adb4086add.png)

Also, is it possible to display custom X and Y labels with the return statement in JavaScript and replace the default labels in the setting area of BB or disable/hide them?

#3 Setup X and Y Axis Label with JavaScript:
![no_own_data_on_xy_axes-Budibase2](https://user-images.githubusercontent.com/2211088/159877746-99ad4acc-4329-4817-96ed-05b907c18bac.png)

#4 Test data for Y Axis Label:
![no_own_data_on_xy_axes-Budibase3](https://user-images.githubusercontent.com/2211088/159877757-9dae34e3-2cf2-449b-a530-ceabd4d9316d.png)

#5 Test data for X Axis Label:
![no_own_data_on_xy_axes-Budibase4](https://user-images.githubusercontent.com/2211088/159884087-d22b74fa-4bdc-43b4-b771-f87254e70523.png)

#6 Unattractive output in the bar chart quadrant:
![no_own_data_on_xy_axes-Budibase](https://user-images.githubusercontent.com/2211088/159877735-bbcad1a3-8a56-4393-b78a-cc3f72599b8e.png)

**Comment by mjashanks** [comment, marked as answer]
Hi @dpedwards 

Charts are definitely in need of some updating.

**Formatting option does not work**

This works for me, however, we have a bug at the moment whereby the chart does not immediately re-render when a setting is changed - you need to refresh the page: https://github.com/Budibase/budibase/issues/5007 - this will be fixed in the next release.

**Axes do not scale correctly**

Agreed - will log this as an issue. I can't see a way around this, other than to display less data or make your chart bigger.

**Setup X and Y Axis Label with JavaScript**

I think you are misunderstanding what this does. It's supposed to label the entire axis, not the actual points on the axis: 

![image](https://user-images.githubusercontent.com/3524181/160373107-f1502754-282b-4462-b058-250f71d44061.png)

**Reply by dpedwards** [reply]
Hi @mjashanks thank you for the detailed feedback. Regarding "Setup X and Y Axis Lable with JavaScript", I mistakenly confused Label Col./Data Cols. with Y Axis Label/X Axis Label. 

![2022-03-28 12_09_54-Budibase](https://user-images.githubusercontent.com/2211088/160377058-f5ea1ace-935d-45ca-89b0-03c45778c470.png)

Y Axis Label or X Axis Label are really only for the axis labels. ;)
![2022-03-28 12_11_23-Budibase](https://user-images.githubusercontent.com/2211088/160376823-02119250-0434-43bf-a8ee-5bd0b852790d.png)
